pandas/core/window/numba_.py

```diff
@@ -161,3 +164,71 @@ def groupby_ewma(
         return result
 
     return groupby_ewma
+
+
+def generate_numba_table_func(
+    args: Tuple,
+    kwargs: Dict[str, Any],
+    func: Callable[..., np.ndarray],
+    engine_kwargs: Optional[Dict[str, bool]],
+    name: str,
+):
+    """
+    Generate a numba jitted function to apply window calculations table-wise.
+
+    Func will be passed a M window size x N number of columns array, and
+    must return a 1 x N number of columns array. Func is intended to operate
+    row-wise, but the result will be transposed for axis=1.
+
+    1. jit the user's function
+    2. Return a rolling apply function with the jitted function inline
+
+    Parameters
+    ----------
+    args : tuple
+        *args to be passed into the function
+    kwargs : dict
+        **kwargs to be passed into the function
+    func : function
+        function to be applied to each window and will be JITed
+    engine_kwargs : dict
+        dictionary of arguments to be passed into numba.jit
+    name : str
+        caller (Rolling/Expanding) and original method name for numba cache key
+
+    Returns
+    -------
+    Numba function
+    """
+    nopython, nogil, parallel = get_jit_arguments(engine_kwargs, kwargs)
+
+    cache_key = (func, f"{name}_table")
+    if cache_key in NUMBA_FUNC_CACHE:
+        return NUMBA_FUNC_CACHE[cache_key]
+
+    numba_func = jit_user_function(func, nopython, nogil, parallel)
+    numba = import_optional_dependency("numba")
+    if parallel:
+        loop_range = numba.prange
+    else:
+        loop_range = range
```

Comment:
Numba aliases `prange` to `range`, if `parallel=False` or the `@njit` decorator is removed it'll still work. Example:
```python
In [1]: from numba import njit, prange

In [2]: @njit
   ...: def foo():
   ...:     for i in prange(4):
   ...:         print(i)
   ...: 

In [3]: foo()
0
1
2
3

In [4]: def foo():
   ...:     for i in prange(4):
   ...:         print(i)
   ...: 

In [5]: foo()
0
1
2
3
```
hope this helps.

Reply:
Thanks! Has this aliasing has been around since version 0.46.0?

Reply:
I think so:
https://github.com/numba/numba/blob/release0.46/numba/special.py#L15-L20